src/Tasks/FileIO/WriteLinesToFile.cs

```diff
@@ -116,17 +121,17 @@ public override bool Execute()
                                 Log.LogMessageFromResources(MessageImportance.Low, "WriteLinesToFile.ErrorReadingFile", File.ItemSpec);
                             }
 
-                            if (contentsAsString == null)
-                            {
-                                contentsAsString = buffer.ToString();
-                            }
 
                             System.IO.File.WriteAllText(File.ItemSpec, contentsAsString, encoding);
                         }
                     }
                     else
                     {
-                        System.IO.File.AppendAllText(File.ItemSpec, buffer.ToString(), encoding);
+                        if (buffer.Length > 0)
```

Comment:
Minimal performance improvement: when the buffer is empty, calling `File.AppendAllText` is a useless IO operation.